src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java

```diff
@@ -35,4 +37,10 @@ public User get(Long id) {
     public Optional<User> findByEmail(String email) {
         return userDao.findByEmail(email);
     }
+
+    @Override
+    public User getUser(Authentication authentication) {
+        UserDetails details = (UserDetails) authentication.getPrincipal();
+        return findByEmail(details.getUsername()).get();
+    }
```

Comment:
this is not part of the application business logic and should be handled on the controller layer

also, have you tried taking `authentication.getName()`?

Reply:
done